src/c_api/c_api.cc

```diff
@@ -1974,7 +1974,8 @@ int MXNDArrayLoad(const char* fname,
         << "Failed to read 32 bits from file.";
   }
 
-  if (magic == 0x04034b50 || magic == 0x504b0304) {  // zip file format; assumed to be npz
+  if (magic == 0x04034b50 || magic == 0x504b0304 || magic == 0x06054b50) {
```

Comment:
You need to consider both big and little endian systems for the magic check here.